tests/TestCase.php

```diff
@@ -46,13 +46,77 @@ abstract class TestCase extends PHPTestCase
 
     protected $rootDir;
 
+    private $catchAllErrorHandler;
+
+    function __construct() {
+        parent::__construct();
+
+        // PHP does not implement setting a class property to an anonymous function,
+        // so we have to do it in the constructor.
+        $this->catchAllErrorHandler = function ($typeNumber, $message, $file, $lineNumber) {
+            $this->fail(
+                sprintf('%s: "%s" in %s:%d',
+                    $this->getErrorType($typeNumber),
+                    $message,
+                    $file,
+                    $lineNumber
+                )
+            );
+        };
+    }
```

Comment:
You can set settings like display_errors in the PHPUnit config as described in https://stackoverflow.com/a/31279986/5301527 for instance.

Reply:
Whoops! The `ini_set('display_errors', 1);` was a leftover from my testing, it's not needed for the test at all! Didn't mean to commit this, sorry about that!

Ok, so this functionality is needed to write a useful test for this issue... Should I revert those changes from this branch and make a new PR for it, and we meanwhile put this PR on-hold until it is merged? I can't think of any other way to really test that this works, other than just letting the test case fail with an error instead of an assertion, which doesn't sound like a sound way to do this to me either.

Reply:
I am not sure if I missed something. Do you mean that you wanna manually provoke an error to see if this thing works?

In that case PHPUnit may help you with functions like `expectException`: https://phpunit.readthedocs.io/en/8.5/writing-tests-for-phpunit.html#testing-php-errors-warnings-and-notices

Reply:
Not quite... The original bug will cause an E_NOTICE with the message `Undefined offset: 1 in smalot/pdfparser/src/Smalot/PdfParser/Parser.php on line 178`. When that happens (e.g. someone might inadvertently reverse the fix, or add some code that will lead to the issue egain, which is what we want to catch with the test), the test case will not fail regularly (with an `F` in the command line and counting towards `Failures:`), but will result in an error (marked with an `E` and resulting in an additional `Error:` count). As far as I can see, following tests are still run, but it doesn't seem to me that this is how an error occuring in a test that we're explicitly trying to prevent should be handled.

By catching all types of errors, the test script doesn't result in an error, but we can mark the test as failed instead if such an error occurs. There's no `expectNotice`, unfortunately, or I would have used that and spared the time! 😄 But I might be thinking overly complicated, and maybe having a test result in an error is just fine!

Reply:
Maybe there is a difference in the **test approach**:

I would start with a test that provokes that `E_NOTICE`. Then you deploy your fix. If you run the same code again, the E_NOTICE should be gone. If it stays, the fix was not complete. Your tests should focus on the things you expect. In this case we expect the E_NOTICE to be gone and all tests run without an `F`. Its hard to test unexpected things. 

Another point, we have these options in the PHPUnit config file: https://github.com/smalot/pdfparser/blob/master/phpunit.xml#L10-L12

```
convertErrorsToExceptions="true"
convertNoticesToExceptions="true"
convertWarningsToExceptions="true"
```
That means that PHPUnit will convert everything fishy to an exception, regardless if its a notice or error. This way you don't have to care about E_NOTICE, because it will be thrown as an exception by PHPUnit all the time.

Reply:
Ok, that's sounds like we're going the `E` route then. I'll remove the complicated testing mechanism and aim for a simpler test. Should a notice pop up in the future, the test will fail as an error. Thanks for your input!